src/java/edu/ucsd/library/dams/api/DAMSAPIServlet.java

```diff
@@ -3375,6 +3306,23 @@ private Map mintDOI( String objid, TripleStore ts, TripleStore es, FileStore fs,
 			xml = obj.getRDFXML(true);
 		}
 
+		// pre-validate
```

Comment:
Does it make sense to extract the validations out into a separate method or DataCiteValidator class? I wonder if these continue to grow if this might get to be a little unwieldy.